### PR TITLE
nix/fossbeamer: cleanup expression and use wrapGAppsHook3

### PR DESCRIPTION
**Pull request by Kranzes**
Follow-up of #32

 - fixes font sizes.
 -  import -> callPackage
 - don't use complex filter src function

### CODE REVIEW

**Comment by flokli** [comment]
Marked as draft, as the parent PR still needs fixes.

**Comment by flokli** [comment]
This also seems to propagate paths to an x86 GTK to the aarch64 binaries, which leads to a big increase in the runtime closure, probably too big for some of the CM3s:

```

/nix/store/0zjl7igj8cfd08ac70qjaiknm23psaqx-morph
└───screen1 -> /nix/store/shz6fg0bl4xbid6028vchpgw6zsjq2f0-nixos-system-fossbeamer-24.11pre-git
    → /nix/store/shz6fg0bl4xbid6028vchpgw6zsjq2f0-nixos-system-fossbeamer-24.11pre-git
    └───activate: …fsx38qi-setup-etc.pl /nix/store/zzhbhsmc4rdw6jvymv6yjpz648bj579p-etc/etc...if (( _localstatus > …
        etc -> /nix/store/zzhbhsmc4rdw6jvymv6yjpz648bj579p-etc/etc
        → /nix/store/zzhbhsmc4rdw6jvymv6yjpz648bj579p-etc
        └───etc/systemd/system -> /nix/store/4wk4jwxgv1awmqb38w2za2gdd34crypp-system-units
            → /nix/store/4wk4jwxgv1awmqb38w2za2gdd34crypp-system-units
            └───cage-tty1.service -> /nix/store/z6qz8m194ik9splbryv6k5x0r4x8jmld-unit-cage-tty1.service/cage-tty1.service
                → /nix/store/z6qz8m194ik9splbryv6k5x0r4x8jmld-unit-cage-tty1.service
                └───cage-tty1.service: …cage \.  '-d' \.  -- /nix/store/xqnk4g81r3dr1h6kajkpvl05n6r1li0x-run-cage-program..IgnoreSIGPIPE…
                    → /nix/store/xqnk4g81r3dr1h6kajkpvl05n6r1li0x-run-cage-program
                    └───/ -> /nix/store/jch3h93cw5ihgggb78q25yr8x1lzf0l0-run-cage-program/bin/run-cage-program
                        → /nix/store/jch3h93cw5ihgggb78q25yr8x1lzf0l0-run-cage-program
                        └───bin/run-cage-program: …nsform 180 --scale 2./nix/store/0rgbi7czjmf7570431k2fqw0ljb437by-rust_fossbeamer-0.1.0-aarch64-u…
                            → /nix/store/0rgbi7czjmf7570431k2fqw0ljb437by-rust_fossbeamer-0.1.0-aarch64-unknown-linux-gnu-linked
                            └───/ -> /nix/store/bzkf764gzqfrsr58mkvjpp0cvh0zkhwr-rust_fossbeamer-0.1.0-aarch64-unknown-linux-gnu
                                → /nix/store/bzkf764gzqfrsr58mkvjpp0cvh0zkhwr-rust_fossbeamer-0.1.0-aarch64-unknown-linux-gnu
                                └───bin/fossbeamer: …desktop-schemas-46.0:/nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42/share/gsettings-s…
                                    → /nix/store/3mzh8lvgbynm9daj7c82k2sfsfhrsfsy-gtk+3-3.24.42
                                    └───lib/libgailutil-3.so.0.0.0: …dmay-glib-2.80.3/lib:/nix/store/l9wi5yjrh39zk0crqi4bzkf4y16nn2lg-at-spi2-core-2.52.0/lib:/nix/st…
                                        lib/libgtk-3.so.0.2410.32: …m8-tracker-3.7.3/lib:/nix/store/l9wi5yjrh39zk0crqi4bzkf4y16nn2lg-at-spi2-core-2.52.0/lib:/nix/st…
                                        → /nix/store/l9wi5yjrh39zk0crqi4bzkf4y16nn2lg-at-spi2-core-2.52.0
                                        └───libexec/.at-spi-bus-launcher-wrapped: …cln-libX11-1.8.9/lib:/nix/store/sbqf71kqhlgaff1ajd8zak09g4wn8ymd-systemd-255.6/lib:/nix/store/m7…
                                            → /nix/store/sbqf71kqhlgaff1ajd8zak09g4wn8ymd-systemd-255.6
                                            └───lib/systemd/systemd-export: …c/static-destruct.h../nix/store/0a0khkw34v25q8k6p44ma1rqa479r1za-gnutar-1.35/bin/tar.Failed to w…
                                                lib/systemd/systemd-import: …subvolume %s: %m...../nix/store/0a0khkw34v25q8k6p44ma1rqa479r1za-gnutar-1.35/bin/tar.Failed to m…
                                                lib/systemd/systemd-pull: …es %s, ignoring: %m../nix/store/0a0khkw34v25q8k6p44ma1rqa479r1za-gnutar-1.35/bin/tar.Failed to r…
                                                → /nix/store/0a0khkw34v25q8k6p44ma1rqa479r1za-gnutar-1.35
```

**Comment by flokli** [comment]
It also fails to execute if deployed to aarch64:

```
Jul 19 12:57:16 fossbeamer cage[6653]: /nix/store/xqnk4g81r3dr1h6kajkpvl05n6r1li0x-run-cage-program: line 3: /nix/store/0rgbi7czjmf7570431k2fqw0ljb437by-rust_fossbeamer-0.1.0-aarch64-unknown-linux-gnu-linked/bin/fossbeamer: cannot execute binary file: Exec format error
```

File: nix/default.nix [hunk 1]
```diff
@@ -26,16 +29,9 @@ let
   ++ gtk3.buildInputs
   ++ pango.buildInputs;
 
-  pkgsAArch64 = import nixpkgs {
-    localSystem = system;
```
Comment (flokli) [inline nix/default.nix:30]: We want to keep this, as we can natively build if invoked on aarch64 (or by passing `--argstr system aarch64-linux` and relying on a remote builder)

File: nix/default.nix [hunk 2]
```diff
@@ -16,6 +18,7 @@ let
     gdk-pixbuf
     glib
     webkitgtk_4_1
+    ./overlay.nix
```
Comment (flokli) [inline nix/default.nix]: :eyes:

Comment (Kranzes) [reply nix/default.nix]: IM A FUCKING IMBECILE AND CANT USE A KEYBOARD :trollface:

File: nix/default.nix [hunk 3]
```diff
@@ -1,13 +1,20 @@
 { system ? builtins.currentSystem
 , nixpkgs ? (import ./sources.nix { inherit system; }).nixpkgs
+, overlays ? [
```
Comment (flokli) [inline nix/default.nix]: why is all this a toplevel arg? It doesn't make a lot of sense, we cannot pass overlays, pkgs or pkgsAarch64.

Can you move it down?

File: nix/overlay.nix [hunk 4]
```diff
@@ -1,38 +0,0 @@
-self: pkgs: rec {
```
Comment (flokli) [inline nix/overlay.nix:1]: Please check `morph build` too, `nix/deployment.nix` also needs to be updated.

**Review by flokli — CHANGES_REQUESTED** [review]

File: nix/default.nix (hunk 3 above)
Comment (Kranzes) [reply nix/default.nix]: brain empty, not a single thought inside.

**Comment by flokli** [comment]
Thanks!

**Pull request merged**